geozero/src/wkt/wkt_writer.rs

```diff
@@ -1,38 +1,42 @@
 use crate::error::Result;
 use crate::{CoordDimensions, FeatureProcessor, GeomProcessor, PropertyProcessor};
 use std::io::Write;
+use std::vec;
 
 use super::WktDialect;
 
 /// WKT Writer.
-pub struct WktWriter<'a, W: Write> {
+pub struct WktWriter<W: Write> {
     dims: CoordDimensions,
     srid: Option<i32>,
     dialect: WktDialect,
     first_header: bool,
-    out: &'a mut W,
+    /// Stack of in-progress geometry sizes
+    geometry_sizes: Vec<usize>,
+    pub(crate) out: W,
 }
 
-impl<'a, W: Write> WktWriter<'a, W> {
-    pub fn new(out: &'a mut W) -> WktWriter<'a, W> {
+impl<W: Write> WktWriter<W> {
+    pub fn new(out: W) -> Self {
```

Comment:
This LGTM, but wanted to highlight the change for other reviewers that we're changing WktWriter to take any `impl Write` rather than only `&mut impl Write`.

Reply:
I'm all for simplifying writters so that they own the underlying `impl Write` - but should it possibly be a separate PR that changes them all?  It feels like a counter-pattern to introduce it in just one writer as part of an unrelated empty geometry support. Then we can rebase this PR on top of it?

Reply:
I'm fine with doing it for all writers. Who wants?

Reply:
I'm ok either way, just as long as we don't release until all writers are more-or-less lined up (and of course its a minor rather than a patch change

Reply:
> and of course its a minor rather than a patch change

Just to be clear, changing from `&mut Write -> Write` is a non-breaking change, right?

Reply:
Yes, there is a breaking change to WktWriter in this PR if that's what you mean (see the changelog entry).

So is everyone OK to merge this?